pkg/secretsencrypt/controller.go

```diff
@@ -223,9 +223,14 @@ func (h *handler) updateSecrets(node *corev1.Node) error {
 	i := 0
 	err = meta.EachListItem(secretsList, func(obj runtime.Object) error {
 		if secret, ok := obj.(*corev1.Secret); ok {
-			if _, err := h.secrets.Update(secret); err != nil {
-				return fmt.Errorf("failed to reencrypted secret: %v", err)
-			}
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				secret, err = h.secrets.Get(secret.Namespace, secret.Name, metav1.GetOptions{})
+				if err != nil {
```

Comment:
What if the secret was deleted? Feels like we should ignore NotFound errors. I'm also having second thoughts about retrying on conflict; if there's a conflict it means that the secret has been updated by something else (and reencrypted in doing so) so we probably don't need to do anything and can just continue on to the next secret.